### ISSUE TITLE
Force close when binding from a Yubikey Neo

### ISSUE THREAD

**Issue by tsujamin** [open-keychain/open-keychain#1315]
On the latest release (3.2.3), and after clearing application data, open-keychain will force close when binding to a newly imported key from a Yubikey Neo.

The key in question is 0x43D41658 (RSA, 4k master, 2k signing, 2k encryption).

I can consistently reproduce it and have submitted a crash report. Phone is running CM12.

**Comment by tsujamin** [comment]
Crash from logcat:

```
E/AndroidRuntime(16529): FATAL EXCEPTION: IntentService[KeychainIntentService]
E/AndroidRuntime(16529): Process: org.sufficientlysecure.keychain, PID: 16529
E/AndroidRuntime(16529): java.lang.NullPointerException: Attempt to invoke virtual method 'long org.spongycastle.openpgp.PGPSecretKey.getKeyID()' on a null object reference
E/AndroidRuntime(16529): 	at org.spongycastle.openpgp.PGPSecretKeyRing.insertSecretKey(PGPSecretKeyRing.java:337)
E/AndroidRuntime(16529): 	at org.sufficientlysecure.keychain.pgp.CanonicalizedPublicKeyRing.createDivertSecretRing(CanonicalizedPublicKeyRing.java:117)
E/AndroidRuntime(16529): 	at org.sufficientlysecure.keychain.operations.PromoteKeyOperation.execute(PromoteKeyOperation.java:87)
E/AndroidRuntime(16529): 	at org.sufficientlysecure.keychain.service.KeychainIntentService.onHandleIntent(KeychainIntentService.java:484)
E/AndroidRuntime(16529): 	at android.app.IntentService$ServiceHandler.handleMessage(IntentService.java:65)
E/AndroidRuntime(16529): 	at android.os.Handler.dispatchMessage(Handler.java:102)
E/AndroidRuntime(16529): 	at android.os.Looper.loop(Looper.java:135)
E/AndroidRuntime(16529): 	at android.os.HandlerThread.run(HandlerThread.java:61)
W/ActivityManager( 2259):   Force finishing activity org.sufficientlysecure.keychain/.ui.ViewKeyActivity
```

**Comment by Valodim** [comment]
huh, I thought I had fixed that when I did c1e7fcf02455b7a03c86bb78efe39684396c15f8. apparently I forgot to commit that change or something.

if you have an android toolchain set up, can you see if it works with current master (after 3be44898db2afd1fd993553e228efaac94f14120)?

**Comment by tsujamin** [comment]
I'll give it a shot in the morning :)

**Comment by Valodim** [comment]
great, thanks

**Comment by tsujamin** [comment]
yeah it appears to be fixed in the current master (5d87872)

**Comment by Valodim** [comment]
aight, cool. will be in the next release :+1:
